### PR TITLE
chore(manifest): add validate boilerplate for rest workload types

### PR DESCRIPTION
**Pull request by iamhopaul123**
Part of #2818.

After this PR, we should have boilerplate completed for every workload type. Then, we'll add more advanced basic types validation as well as start to consolidate validation elsewhere.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License..

### CODE REVIEW

File: internal/pkg/manifest/validate.go [hunk 1]
```diff
@@ -44,9 +44,132 @@ func (l *LoadBalancedWebServiceConfig) Validate() error {
...
+	return nil
+}
+
+// Validate returns if BackendServiceConfig is configured correctly.
+func (b *BackendServiceConfig) Validate() error {
+	var err error
+	if err = b.ImageConfig.Validate(); err != nil {
+		return fmt.Errorf(`validate "image": %w`, err)
+	}
+	if err = b.ImageOverride.Validate(); err != nil {
```
Comment (huanjani) [inline internal/pkg/manifest/validate.go:61]: Just curious why some errors are wrapped and others (here and below) aren't.

File: internal/pkg/manifest/validate.go [hunk 2]
```diff
@@ -44,9 +44,132 @@ func (l *LoadBalancedWebServiceConfig) Validate() error {
...
-			return fmt.Errorf(`validate taskdef_overrides[%d]: %w`, i, err)
+	for ind, taskDefOverride := range l.TaskDefOverrides {
+		if err = taskDefOverride.Validate(); err != nil {
+			return fmt.Errorf(`validate taskdef_overrides[%d]: %w`, ind, err)
+		}
+	}
+	return nil
+}
+
+// Validate returns if BackendServiceConfig is configured correctly.
```
Comment (huanjani) [inline internal/pkg/manifest/validate.go]: nit: these comments (here and below) make it sound like these methods return bools. Maybe: 
```suggestion
// Validate returns nil if BackendServiceConfig is configured correctly.
```
or some other rewording?

File: internal/pkg/manifest/validate.go (hunk 1 above)
Comment (iamhopaul123) [reply internal/pkg/manifest/validate.go:61]: It depends on if the struct is `inlined` or not (for better error msg helping them to debug)

File: internal/pkg/manifest/validate.go (hunk 2 above)
Comment (iamhopaul123) [reply internal/pkg/manifest/validate.go]: good idea!

File: internal/pkg/manifest/validate.go [hunk 3]
```diff
@@ -95,25 +224,28 @@ func (i *Image) Validate() error {
 	return nil
 }
 
-// Validate returns if BuildArgsOrString is configured correctly.
+// Validate returns nil if BuildArgsOrString is configured correctly.
 func (b *BuildArgsOrString) Validate() error {
+	if b.isEmpty() {
```
Comment (efekarakus) [inline internal/pkg/manifest/validate.go:229]: nice! i like this change of moving it inside, simplifies it for the callers.

**Review by efekarakus — APPROVED** [review]

**Review by huanjani — APPROVED** [review]
LGTM!

**Pull request merged**